### PR TITLE
Update demangler with fixes from FreeBSD and elftoolchain

### PR DESCRIPTION
**Pull request by DimitryAndric**
Brings the elftoolchain-based demangler up-to-date with recent changes.

### CODE REVIEW

File: src/libelftc_dem_gnu3.c [hunk 1]
```diff
@@ -127,7 +153,7 @@ get_strlen_sum(const struct vector_str *v)
 /**
  * @brief Deallocate resource in vector_str.
  */
-static void
+void
```
Comment (davidchisnall) [inline src/libelftc_dem_gnu3.c:156]: Why are we losing `static` on these?  I don't think we want to expose any of them outside this compilation unit.

**Review by davidchisnall — COMMENTED** [review]
Looks fine, but I'm slightly concerned that losing the `static` on some of the functions impedes inlining and will generate worse code.  Is there a reason for that change?

File: src/libelftc_dem_gnu3.c (hunk 1 above)
Comment (DimitryAndric) [reply src/libelftc_dem_gnu3.c:156]: Oh I didn't intentionally remove them, just copied all the code from the upstream `libelftc_vstr.c` file. This was one of the changes I didn't realize were in there.

Comment (davidchisnall) [reply src/libelftc_dem_gnu3.c:156]: If the code size doesn't get worse without `static`, I'm happy to lose them to minimise diffs from upstream.  If it does, I think I'd like them brought back.

Comment (DimitryAndric) [reply src/libelftc_dem_gnu3.c]: I just re-added the static keywords.

**Review by davidchisnall — APPROVED** [review]

**Comment by davidchisnall** [comment]
Thanks!

**Pull request merged**